src/Database/Adapter/SQL.php

```
@@ -981,7 +991,7 @@ protected function getSQLPermissionsCondition(string $collection, array $roles):
                     SELECT _document
                     FROM {$this->getSQLTable($collection . '_perms')}
                     WHERE _permission IN (" . implode(', ', $roles) . ")
-                      AND _type = 'read'
```

Because this can come from outside the library and we're not binding it, let's throw an exception if $type is not in the `Database::PERMISSIONS` array